static/src/javascripts/projects/membership/contributions-recurring-tab.js

```diff
@@ -0,0 +1,192 @@
+};
+
+const hidePayPalAccountName = (): void => {
+    $(PAYPAL_SHOW_EMAIL_MESSAGE).addClass(IS_HIDDEN_CLASSNAME);
+    $(PAYPAL_SHOW_EMAIL_BUTTON).removeClass(IS_HIDDEN_CLASSNAME);
+    $(PAYPAL_HIDE_EMAIL_BUTTON).addClass(IS_HIDDEN_CLASSNAME);
+};
+
+const showPayPalAccountName = (): void => {
+    $(PAYPAL_SHOW_EMAIL_MESSAGE).removeClass(IS_HIDDEN_CLASSNAME);
+    $(PAYPAL_HIDE_EMAIL_BUTTON).removeClass(IS_HIDDEN_CLASSNAME);
+    $(PAYPAL_SHOW_EMAIL_BUTTON).addClass(IS_HIDDEN_CLASSNAME);
+};
+
+const displayContributionInfo = (): void => {
+    $(CONTRIBUTION_INFO).removeClass(IS_HIDDEN_CLASSNAME);
+};
+
+const displaySupportUpSell = (): void => {
+    $(UP_SELL).removeClass(IS_HIDDEN_CLASSNAME);
+};
+
+const displayErrorMessage = (): void => {
+    $(ERROR).removeClass(IS_HIDDEN_CLASSNAME);
+};
+
+const populateUserDetails = (contributorDetails: ContributorDetails): void => {
+    const isMonthly = contributorDetails.subscription.plan.interval === 'month';
+    const intervalText = isMonthly ? 'Monthly' : 'Annual';
+    const glyph = contributorDetails.subscription.plan.currency;
+    let notificationTypeSelector;
+
+    // $(MEMBERSHIP_TIER).text(contributorDetails.tier);
+    // $(PACKAGE_COST).text(
+    //     formatters.formatAmount(contributorDetails.subscription.plan.amount, glyph)
+    // );
+    // $(DETAILS_JOIN_DATE).text(formatters.formatDate(contributorDetails.joinDate));
+    $(PACKAGE_INTERVAL).text(intervalText);
+
+    const cardTail =
+        contributorDetails.subscription.card &&
+        contributorDetails.subscription.card.last4;
+    if (cardTail) {
+        $(MANAGE_CARD_LAST4).text(cardTail);
+    } else if (contributorDetails.subscription.payPalEmail) {
+        $(PAYPAL_EMAIL_ADDRESS).text(
+            contributorDetails.subscription.payPalEmail
+        );
+    }
+
+    // $(PACKAGE_CURRENT_PERIOD_END).text(
+    //     formatters.formatDate(contributorDetails.subscription.end)
+    // );
+    // $(PACKAGE_CURRENT_RENEWAL_DATE).text(
+    //     formatters.formatDate(contributorDetails.subscription.renewalDate)
+    // );
+
+    if (contributorDetails.subscription.nextPaymentDate) {
+        $(PACKAGE_NEXT_PAYMENT_DATE).text(
+            formatters.formatDate(
+                contributorDetails.subscription.nextPaymentDate
+            )
+        );
+        // $(PACKAGE_NEXT_PAYMENT_CONTAINER).removeClass(IS_HIDDEN_CLASSNAME);
+    }
+
+    $(PACKAGE_NEXT_PAYMENT_PRICE).text(
+        formatters.formatAmount(
+            contributorDetails.subscription.plan.amount,
+            glyph
+        )
+    );
+
+    // if (contributorDetails.subscription.trialLength > 0) {
+    //     $(TRIAL_INFO_CONTAINER).removeClass(IS_HIDDEN_CLASSNAME);
+    // }
+
+    // display membership number
+    if (contributorDetails.subscription.subscriberId) {
+        $(DETAILS_SUBSCRIPTION_NUMBER_CONTAINER).removeClass(
+            IS_HIDDEN_CLASSNAME
+        );
+        $(DETAILS_SUBSCRIPTION_NUM_TEXT).text(
```

Comment:
Personally I would prefer to move away from exposing subscription names/numbers to users - but I guess UX will make the final call on that!

Reply:
In terms of an internal reference number, the users email address or identity id should be sufficient for userhelp to identify their regular contribution in Zuora

Reply:
Agreed. Elvis and I had a conversation with user help yesterday and arrived at the same conclusion, so we've removed the Contributor ref :)